File: fsharp/parallel-letter-frequency/ParallelLetterFrequency.fs
module ParallelLetterFrequency

let frequency texts =
    let frequencyAsync (text: string) =
        async {
            return text.ToLower()
            |> Seq.filter System.Char.IsLetter
            |> Seq.countBy id
        }

    texts
    |> List.map frequencyAsync
    |> Async.Parallel
    |> Async.RunSynchronously
    |> Seq.concat
    |> Seq.groupBy fst
    |> Seq.map (fun (letter, occurences) -> letter, occurences |> Seq.sumBy snd)
    |> Map.ofSeq

File: csharp/parallel-letter-frequency/ParallelLetterFrequency.cs
using System;
using System.Collections.Concurrent;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

public static class ParallelLetterFrequency
{
    public static Dictionary<char, int> Calculate(IEnumerable<string> texts)
    {
        var freqMap = new ConcurrentDictionary<char, int>();

        Parallel.ForEach(texts, text =>
        {
            foreach (var ch in text.ToLower().Where(char.IsLetter))
            {
                freqMap.AddOrUpdate(ch, 1, (k, v) => v + 1);
            }
        });

        return new Dictionary<char, int>(freqMap);
    }
}